pep-9999.rst

```diff
@@ -0,0 +1,709 @@
+checkers, linters or language servers. They are also costly to
+maintain for library authors, and they require that Python developers
+know about the existence of these plugins and download and configure
+them within their environment.
+
+
+Rationale
+=========
+The intent of this proposal is not to support every feature of every
+library with dataclass-like semantics, but rather to make it possible
+to use the most common features of these libraries in a way that is
+compatible with static type checking. If a user values these libraries
+and also values static type checking, they may need to avoid using
+certain features or make small adjustments to the way they use these
+libraries. That's already true for the mypy custom plugins, which
+don't support every feature in many of these libraries.
+
+
+Specification
+=============
+
+The ``dataclass_transform`` decorator
+-------------------------------------
+This specification introduces a new decorator function exported from
+the ``typing`` module named ``dataclass_transform``. This decorator
+can be applied to either a function (which is a decorator function
+itself), a class, or a metaclass. The presence of
+``dataclass_transform`` tells a static type checker that the decorated
+function, class, or metaclass performs runtime "magic" that transforms
+a class, endowing it with dataclass-like behaviors.
+
+If ``dataclass_transform`` is applied to a function, the use of this
+function as a decorator is assumed to apply dataclass-like semantics.
+If ``dataclass_transform`` is applied to a class, dataclass-like
+semantics will be assumed for any class that derives from the
+decorated class or uses the decorated class as a metaclass.
+
+Examples of each approach are shown in the following sections. Each
+example creates a ``CustomerModel`` class with dataclass-like semantics.
+Type checkers supporting this PEP will recognize that the
+``CustomerModel`` class can be instantiated using the synthesized
+``__init__`` method::
+
+  # Using positional arguments
+  c1 = CustomerModel(327, "John Smith")
+
+  # Using keyword arguments
+  c2 = CustomerModel(id=327, name="John Smith")
+
+  # These calls will generate runtime errors and should be flagged as
+  # errors by a static type checker.
+  c3 = CustomerModel()
+  c4 = CustomerModel(327, first_name="John")
+  c5 = CustomerModel(327, "John Smith", 0)
+
+Decorator function example
+``````````````````````````
+Here is an example of using ``dataclass_transform`` to decorate a
+decorator function named ``create_model``. The implementation of the
+``create_model`` function is omitted for brevity, but we assume here
+that it modifies the class it decorates in the following ways:
+
+1. It synthesizes an ``__init__`` method using data fields declared
+   within the class and its parent classes.
+2. It synthesizes an ``__eq__`` and ``__ne__`` method.
+
+::
+
+  # The `create_model` decorator is defined by a library. This could
```

Comment:
```suggestion
  # The ``create_model`` decorator is defined by a library. This could
```
The RST linter wants this to be two backticks (applies to a bunch of other lines below).

Reply:
(Not to mention that it won't render correctly without it, which is why it is complaining :)